src/ui/widgets/tui_dirlist.rs

```diff
@@ -90,9 +90,9 @@ fn print_entry(
         }
         _ => {
             let file_drawing_width = drawing_width;
-            let (stem, extension) = match name.rfind('.') {
-                None => (name, ""),
-                Some(i) => name.split_at(i),
+            let (stem, extension) = match entry.get_ext() {
```

Comment:
I think rather than changing `get_ext`'s behaviour, its better we add a case here instead.
Something like:
```rust
let (stem, extension) = match name.rfind('.') {
    None | Some(0) => (name, ""),
    Some(i) => name.split_at(name.len() - ext.len()),
};
```
Hopefully, this still works as intended xd

Reply:
Unfortunately this will not work because of devicons: `name` is actually the entry's label:
https://github.com/kamiyaa/joshuto/blob/f42660c2255a5db1a6c0918ac790bee7fae72d80/src/ui/widgets/tui_dirlist.rs#L80